### DISCUSSION TITLE
remount a pvc volume

### DISCUSSION THREAD

**Discussion by gaord** [rook/rook#13091] [Q&A]
Hi, I would like to remount a pvc (to refresh cephfs client) consumed by a pod in linux host level. What will be safe steps to achieve that? I can stop a pod for short time, however its pvc still gets mounted/bound. Any clues to remount a volume without touching the content inside the volume?

rook: 1.10.10

**Comment by travisn** [comment]
What is the goal? I am not clear on your question. Do you want to mount the volume in another pod? Or you want to do maintenance by stopping the application and ensure the volume is still there when the application comes back online?

**Reply by gaord** [reply]
we got warning from ceph cluster. one of cephfs client is not behaving properly as should be. This warning has negative impact on storage performance, observed. Currently, we only have option to restart storage client as a solution, that in fact is to get related volume unmount and mount again on the linux node level. We had tried to unmount/mount volume manually, rook seemed get stuck afterwards in provisioning operation, however. We went around it by restarting all rook pods anyway. For rook provisions the storage in k8s, I wonder if rook could help to do that. Or any other ideas? Thanks.

here is ceph warning:
ceph health detail
[WRN] MDS_CLIENT_LATE_RELEASE: 1 clients failing to respond to capability release
    mds.code-store.host17w.pdziet(mds.0): Client k8s-33 failing to respond to capability release client_id: 18618895

here is log showing rook stuck in provisioning after manually mounting the volume:
Events:
  Type     Reason       Age                   From               Message
  ----     ------       ----                  ----               -------
  Normal   Scheduled    5m25s                 default-scheduler  Successfully assigned devops/10.168.37.161-d4kdz to k8s-28
  Warning  FailedMount  3m22s                 kubelet            Unable to attach or mount volumes: unmounted volumes=[volume-0], unattached volumes=[default-token-7blsf volume-0 volume-1 workspace-volume]: timed out waiting for the condition
  Warning  FailedMount  102s (x2 over 3m44s)  kubelet            MountVolume.MountDevice failed for volume "pvc-3c2f99a7-8ac6-4f67-93af-d7919cafaf1a" : rpc error: code = Aborted desc = an operation with the given Volume ID 0001-0012-rook-ceph-external-0000000000000001-3a69d51a-f3cd-11ed-b738-964ec15fdba7 already exists
  Warning  FailedMount  66s                   kubelet            Unable to attach or mount volumes: unmounted volumes=[volume-0], unattached volumes=[volume-0 volume-1 workspace-volume default-token-7blsf]: timed out waiting for the condition

**Reply by travisn** [reply]
The CSI driver is expected to unmount the volume, otherwise the driver thinks the volume is still mounted. 
@Rakshith-R For a workaround, does the volume attachment for that volume just need to be deleted?

**Reply by Madhu-1** [reply]
deleting volume attachment will not work as the real `umount` might be stuck, The above warning is due to the failure to attach the volume and it could be due to ceph not responding problem.  The best option is to bring ceph status to OK state and restart the csi-cephfsplugin pod. https://github.com/ceph/ceph-csi/issues/3657 might help to an extent if ceph calls are stuck.

**Reply by gaord** [reply]
thanks for inputs on this.
to put it simple, for pvc-000e2041-880c-419f-b933-d50babd9cb6b, I would do this to fix an issue:
1,stop all pods attaching it, 
2, umount  /apps/kubernetes/plugins/kubernetes.io/csi/pv/pvc-000e2041-880c-419f-b933-d50babd9cb6b/globalmount
umount /apps/kubernetes/pods/59337fc3-6cfd-4cf2-9d7c-4c624bfc640b/volumes/kubernetes.io~csi/pvc-000e2041-880c-419f-b933-d50babd9cb6b/mount
3, start all pods

would rook/storage provisioning will handle it smoothly? Any other ideas?
 
here is host mounts:
$ mount|grep cb6b
169.169.82.57:6789,169.169.7.81:6789,169.169.241.153:6789:/volumes/csi/csi-vol-c721d7d5-1a36-11ee-85e1-4a507c3944b9/f5c316b0-5391-4022-92d1-b0646427fa77 on /apps/kubernetes/plugins/kubernetes.io/csi/pv/pvc-000e2041-880c-419f-b933-d50babd9cb6b/globalmount type ceph (rw,relatime,name=csi-cephfs-node,secret=<hidden>,fsid=00000000-0000-0000-0000-000000000000,acl,mds_namespace=myfs,_netdev)
169.169.82.57:6789,169.169.7.81:6789,169.169.241.153:6789:/volumes/csi/csi-vol-c721d7d5-1a36-11ee-85e1-4a507c3944b9/f5c316b0-5391-4022-92d1-b0646427fa77 on /apps/kubernetes/pods/59337fc3-6cfd-4cf2-9d7c-4c624bfc640b/volumes/kubernetes.io~csi/pvc-000e2041-880c-419f-b933-d50babd9cb6b/mount type ceph (rw,relatime,name=csi-cephfs-node,secret=<hidden>,fsid=00000000-0000-0000-0000-000000000000,acl,mds_namespace=myfs,_netdev)

**Reply by gaord** [reply]
it looks like to scale the deployment to 0 will do the work. Thanks

**Comment by microyahoo** [comment]
Do you mean reusing existing pv like this?
https://github.com/kubernetes/kubernetes/issues/48609
